### PR TITLE
Batch of updates for the image

### PR DESCRIPTION
**Pull request by dduportal**
Hello there !

Follow up of #11 without alpine.

After some time using my own image which purpose is almost the same as this one, here is an update proposal that cover an upgrade to latest docker practises (1.13.2 at the time starting this PR), Jenkins SSH usages, and also shrinking the image to spend less disk space.

Here are the major changes:

* Base image moved from "[java](https://hub.docker.com/_/java/)" (deprecated in 2016) to "[openjdk](https://hub.docker.com/_/openjdk/)"
* No more slave.jar (test upgraded) embeded since the Jenkins SSH agent protocol takes care of downloading and upgrading it at startup
* Ensuring that bash is the default command line (cf. https://issues.jenkins-ci.org/browse/JENKINS-41374 )
* Adapting Dockerfile instructions to use builds ARGS to allow specification of user and groups, uid and gid, following [the official Jenkins image](https://github.com/jenkinsci/docker/blob/master/Dockerfile#L8)
* SSH is now writing its logs to stderr, allowing docker logs to fetch the SSH output (and not written within a file)
* Additional data volume have been added to improve agent performances (writing temp files on host FS and not on the Layered FS)
* The default user home directory is now variabilized for better maintenance or custom build use cases
* git is now explicitly installed (tradeoff because it generally require root rights to install, so interesting to have it here, while mercurial don't)
* Some bash shellchecking

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
@reviewbybees

File: Dockerfile [hunk 1]
```diff
@@ -20,24 +20,35 @@
...
-WORKDIR /home/jenkins
+RUN groupadd -g ${gid} ${group} \
+    && useradd -d "${JENKINS_AGENT_HOME}" -u "${uid}" -g "${gid}" -m -s /bin/bash "${user}"
 
 # setup SSH server
-RUN apt-get update && apt-get install -y openssh-server
-RUN sed -i 's/#PasswordAuthentication yes/PasswordAuthentication no/' /etc/ssh/sshd_config
+RUN apt-get update \
+    && apt-get install --no-install-recommends -y \
+        openssh-server git \
```
Comment (oleg-nenashev) [inline Dockerfile]: I am not convinced Git is required in the image.
Good for demos, but Jenkins has tool installers which can install a portable Git client

File: setup-sshd [hunk 2]
```diff
@@ -28,10 +30,10 @@
...
 
 write_key() {
-	mkdir -p /home/jenkins/.ssh
-	echo "$1" > /home/jenkins/.ssh/authorized_keys
-	chown -Rf jenkins:jenkins /home/jenkins/.ssh
-	chmod 0700 -R /home/jenkins/.ssh
+	mkdir -p "${JENKINS_AGENT_HOME}/.ssh"
+	echo "$1" > "${JENKINS_AGENT_HOME}/.ssh/authorized_keys"
+	chown -Rf jenkins:jenkins "${JENKINS_AGENT_HOME}/.ssh"
+	chmod 0700 -R "${JENKINS_AGENT_HOME}/.ssh"
```
Comment (oleg-nenashev) [inline setup-sshd]: I will follow-up in the PM

File: tests/test_helpers.bash [hunk 3]
```diff
@@ -1,4 +1,4 @@
-
+#!/bin/bash
```
Comment (oleg-nenashev) [inline tests/test_helpers.bash]: "-ex" as well?

File: Dockerfile (hunk 1 above)
Comment (dduportal) [inline Dockerfile]: Updated

File: tests/test_helpers.bash (hunk 3 above)
Comment (dduportal) [inline tests/test_helpers.bash]: Gotcha, updated :)

File: setup-sshd (hunk 2 above)
Comment (dduportal) [inline setup-sshd]: Not sure to understand this ?

**Comment by dduportal** [comment]
Git removed, and secured the bash helper for tests.

Not sure to get WDYM by the PM follow up (what is the PM acronym for ? )

**Pull request merged**